packages/data/src/namespace-store/index.js

```diff
@@ -26,9 +26,11 @@ import * as metadataActions from './metadata/actions';
 /**
  * Creates a namespace object with a store derived from the reducer given.
  *
- * @param {string} key              Identifying string used for namespace and redex dev tools.
- * @param {Object} options          Contains reducer, actions, selectors, and resolvers.
- * @param {Object} registry         Registry reference.
+ * @param {string}         key      Unique namespace identifier.
+ * @param {Object}         options  Registered store options, with properties
+ *                                  describing reducer, actions, selectors, and
+ *                                  resolvers.
+ * @param {WPDataRegistry} registry Registry reference.
```

Comment:
There's no way for editors to know how to interpret this without importing it in the file as its own typedef.

Reply:
> There's no way for editors to know how to interpret this without importing it in the file as its own typedef.

Hm, is this what the `import` is used for in your own pull requests? How does that work with `@typedef` (vs. your TypeScript definitions)?

Reply:
Precisely.

In javascript, in order to reference types that are defined elsewhere (be it in a typescript definition file, or as standard JSDoc), the types must first be imported in the referencing file as a `@typedef`. Once imported, you can freely reference the type throughout that file.

Otherwise the editor will not be able to provide type hints or "intellisense" (in VS editors)